managers/asset-operations-manager.js

```diff
@@ -344,21 +344,45 @@ class AssetOperationsManager {
         );
 
         const { tokenId } = resolveUAL(UAL);
+
         let hasPendingUpdate = false;
         if (state === ASSET_STATES.LATEST) {
             hasPendingUpdate = await this.blockchainService.hasPendingUpdate(tokenId, blockchain);
         }
 
-        const publicAssertionId = hasPendingUpdate
-            ? await this.blockchainService.getUnfinalizedState(tokenId, blockchain)
-            : await this.blockchainService.getLatestAssertionId(tokenId, blockchain);
+        let publicAssertionId;
+        if (hasPendingUpdate) {
+            publicAssertionId = await this.blockchainService.getUnfinalizedState(tokenId, blockchain);
+        } else {
+            const assertionIds = await this.blockchainService.getAssertionIds(tokenId, blockchain);
+
+            if (state === ASSET_STATES.FINALIZED) {
+                publicAssertionId = assertionIds.slice(-1);
```

Comment:
I might be misunderstanding the chart, but slice(-1) returns an array containing the last element and I think according to the chart we want just the last element. At least this line and line 369 should be in the same format (either element or subarray containing it) according to the chart

Reply:
You're right, I fixed it